camel-route-support/src/main/java/io/kaoto/backend/api/service/deployment/generator/camelroute/CamelRouteDeploymentGeneratorService.java

```diff
@@ -117,8 +125,28 @@ public Pod getPod(final String namespace, final String name, final KubernetesCli
         return null;
     }
 
+
     @Override
-    public Stream<Step> filterCatalog(String previousStep, String followingStep, Stream<Step> steps) {
+    public Stream<Step> filterCatalog(Step previousStep, Step followingStep, Stream<Step> steps) {
+
+        if (previousStep != null) {
+            switch (previousStep.getKind().toUpperCase()) {
+                case "CAMEL-REST-DSL":
+                    //After the "rest" element, comes a http verb
+                    steps = steps.filter(s -> s.getKind().equalsIgnoreCase("CAMEL-REST-VERB"));
```

Comment:
Maybe worth using constants instead of repeating strings?

Reply:
Yes, doing it.